recipes/boost/all/conanfile.py

```diff
@@ -691,6 +691,9 @@ def _gnu_cxx11_abi(self):
     def _build_flags(self):
         flags = self._build_cross_flags
 
+        # Stop at the first error. No need to continue building.
+        flags.append("-q")
```

Comment:
If this was just for testing, I would like to see this kept, will help CCI

Reply:
I added it intentionally to fail faster and not dig into the logs after the fact.
I hoped it would fail when some option could not be satisfied, but it doesn't do that.

So, even when `--with-locale` is passed, it will not be built when iconv/icu cannot be found. No error is thrown.